### PR TITLE
Split resource struct to dynamic and static ones

### PR DESCRIPTION
**Pull request by anttiylitokola**

### CODE REVIEW

File: nsdl-c/sn_nsdl_lib.h [hunk 1]
```diff
@@ -135,53 +135,41 @@ typedef enum sn_nsdl_resource_mode_ {
...
-
-    uint16_t    coap_content_type;
-//    uint8_t     mime_content_type;
-
-    uint8_t     *resource_type_ptr;
-    uint8_t     *interface_description_ptr;
-
-} sn_nsdl_resource_parameters_s;
+    uint16_t    coap_content_type;          // This can be removed?
+    uint8_t     access:4;
```
Comment (TeroJaasko) [inline nsdl-c/sn_nsdl_lib.h]: At some point this access could be moved to be after the pointers to save 1 byte of padding.

File: source/libNsdl/src/sn_grs.c [hunk 2]
```diff
@@ -344,27 +362,147 @@ extern int8_t sn_grs_create_resource(struct grs_s *handle, sn_nsdl_resource_info
...
+            return SN_NSDL_FAILURE;
+        }
+
+        memset(resource_copy_ptr->static_resource_parameters, 0, sizeof(sn_nsdl_static_resource_parameters_s));
+        resource_copy_ptr->static_resource_parameters->observable = resource_ptr->static_resource_parameters->observable;
+        resource_copy_ptr->static_resource_parameters->mode = resource_ptr->static_resource_parameters->mode;
+        resource_copy_ptr->static_resource_parameters->external_memory_block = resource_ptr->static_resource_parameters->external_memory_block;
+        resource_copy_ptr->static_resource_parameters->access = resource_ptr->static_resource_parameters->access;
+        resource_copy_ptr->static_resource_parameters->coap_content_type = resource_ptr->static_resource_parameters->coap_content_type;
+
```
Comment (TeroJaasko) [inline source/libNsdl/src/sn_grs.c]: As a future dev suggestion, temporary variables caching the value of "resource_copy_ptr->static_resource_parameters" and "resource_ptr->static_resource_parameters" might reduce generated code. Or compiler may be able to optimize that itself. At least the source lines might fit to screen;)

File: source/libNsdl/src/sn_grs.c [hunk 3]
```diff
@@ -441,71 +579,27 @@ extern int8_t sn_grs_process_coap(struct nsdl_s *nsdl_handle, sn_coap_hdr_s *coa
...
-                                }
-                                memcpy(resource_temp_ptr->resource, coap_packet_ptr->payload_ptr, resource_temp_ptr->resourcelen);
-                            }
-                            if (coap_packet_ptr->content_format != COAP_CT_NONE) {
-                                if (resource_temp_ptr->resource_parameters_ptr) {
-                                    resource_temp_ptr->resource_parameters_ptr->coap_content_type = coap_packet_ptr->content_format;
-                                }
-                            }
-                            status = COAP_MSG_CODE_RESPONSE_CHANGED;
-                        } else {
```
Comment (anttiylitokola) [inline source/libNsdl/src/sn_grs.c]: This part is for static resource handling, agreed with Yogesh that only GET should be supported in this case.

**Comment by TeroJaasko** [comment]
+1

**Pull request merged**